### PR TITLE
Add a 400ZR test for module temperature reporting.

### PR DESCRIPTION
**Pull request by ahsaanyousaf**

### CODE REVIEW

File: feature/platform/transceiver/zr_temperature/README.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    following ZR transceiver telemetry paths exist and are streamed for both
+    the ZR optics.
+    *   /platform/components/component/state/temperature/instant
+    *   /platform/components/component/state/temperature/min
+    *   /platform/components/component/state/temperature/max
+    *   /platform/components/component/state/temperature/avg
+*   For reported data check for validity min <= avg/instant <= max
+
+*   If the modules or the devices are in a boot stage, they must not stream
+    any invalid string values like "nil" or "-inf" until valid values
```
Comment (ejbrever) [inline feature/platform/transceiver/zr_temperature/README.md:31]: you can remove the "until valid values are available for streaming" as this reads that later during steady state that these invalid values are okay :)

File: feature/platform/transceiver/zr_temperature/README.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    *   Use /interfaces/interface/config/enabled to disable the interfaces and
+        wait 120 seconds(cooling off period) before taking the temperature
+        reading again.
+    *   Verify the module is able to stream the temperature data in this state.
+    *   Verify the module reported temperature in this state is always less
+        than the module temperature captured during steady state operation with
+        interface state enabled.
+    *   For reported data check for validity min <= avg/instant <= max
+    
+    *   If the modules or the devices are in a boot stage, they must not stream
```
Comment (ejbrever) [inline feature/platform/transceiver/zr_temperature/README.md:54]: similar to above, please remove "until valid values are available for streaming"

Comment (ahsaanyousaf) [reply feature/platform/transceiver/zr_temperature/README.md:53]: Done

File: feature/platform/transceiver/zr_temperature/README.md (hunk 1 above)
Comment (ahsaanyousaf) [reply feature/platform/transceiver/zr_temperature/README.md]: Done

**Review by dplore — APPROVED** [review]

**Review by dplore — CHANGES_REQUESTED** [review]
Please add entry to testregistry.textproto

**Review by dplore — APPROVED** [review]

**Pull request merged**